File: 1813-sentence-similarity-iii/1813-sentence-similarity-iii.py
class Solution:
    def areSentencesSimilar(self, sentence1: str, sentence2: str) -> bool:
        # Split the sentences into lists of words
        words1 = sentence1.split()
        words2 = sentence2.split()

        # Ensure words1 is the longer list
        if len(words1) < len(words2):
            words1, words2 = words2, words1

        len_words1 = len(words1)
        len_words2 = len(words2)

        # Check for matching from the start
        start_index = 0
        while start_index < len_words2 and words1[start_index] == words2[start_index]:
            start_index += 1

        # Check for matching from the end
        end_index = 0
        while end_index < len_words2 and words1[len_words1 - 1 - end_index] == words2[len_words2 - 1 - end_index]:
            end_index += 1

        # Check if the unmatched part of the longer sentence can accommodate the unmatched part of the shorter sentence
        return start_index + end_index >= len_words2

File: 1813-sentence-similarity-iii/1813-sentence-similarity-iii.js
/**
 * @param {string} sentence1
 * @param {string} sentence2
 * @return {boolean}
 */
var areSentencesSimilar = function(sentence1, sentence2) {
    // Split the sentences into arrays of words
    let words1 = sentence1.split(' ');
    let words2 = sentence2.split(' ');

    // Ensure words1 is the longer array
    if (words1.length < words2.length) {
        [words1, words2] = [words2, words1]; // Swap using destructuring assignment
    }

    const lenWords1 = words1.length;
    const lenWords2 = words2.length;

    // Check for matching from the start
    let startIndex = 0;
    while (startIndex < lenWords2 && words1[startIndex] === words2[startIndex]) {
        startIndex++;
    }

    // Check for matching from the end
    let endIndex = 0;
    while (endIndex < lenWords2 && words1[lenWords1 - 1 - endIndex] === words2[lenWords2 - 1 - endIndex]) {
        endIndex++;
    }

    // Check if the unmatched part of the longer sentence can accommodate the unmatched part of the shorter sentence
    return startIndex + endIndex >= lenWords2;
};